src/test/java/com/commercetools/sync/sdk2/producttypes/utils/AttributeDefinitionUpdateActionUtilsTest.java

```diff
@@ -0,0 +1,840 @@
+import com.commercetools.api.models.product_type.AttributePlainEnumValueBuilder;
+import com.commercetools.api.models.product_type.AttributeTypeBuilder;
+import com.commercetools.api.models.product_type.ProductTypeAddLocalizedEnumValueActionBuilder;
+import com.commercetools.api.models.product_type.ProductTypeAddPlainEnumValueActionBuilder;
+import com.commercetools.api.models.product_type.ProductTypeChangeAttributeConstraintActionBuilder;
+import com.commercetools.api.models.product_type.ProductTypeChangeInputHintActionBuilder;
+import com.commercetools.api.models.product_type.ProductTypeChangeIsSearchableActionBuilder;
+import com.commercetools.api.models.product_type.ProductTypeChangeLabelActionBuilder;
+import com.commercetools.api.models.product_type.ProductTypeChangeLocalizedEnumValueLabelActionBuilder;
+import com.commercetools.api.models.product_type.ProductTypeChangeLocalizedEnumValueOrderActionBuilder;
+import com.commercetools.api.models.product_type.ProductTypeChangePlainEnumValueLabelActionBuilder;
+import com.commercetools.api.models.product_type.ProductTypeChangePlainEnumValueOrderActionBuilder;
+import com.commercetools.api.models.product_type.ProductTypeRemoveEnumValuesActionBuilder;
+import com.commercetools.api.models.product_type.ProductTypeSetInputTipActionBuilder;
+import com.commercetools.api.models.product_type.ProductTypeUpdateAction;
+import com.commercetools.api.models.product_type.TextInputHint;
+import com.commercetools.sync.sdk2.producttypes.helpers.ResourceToDraftConverters;
+import java.util.List;
+import java.util.Optional;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class AttributeDefinitionUpdateActionUtilsTest {
+  private static AttributeDefinition old;
+  private static AttributeDefinitionDraft newSame;
+  private static AttributeDefinitionDraft newDifferent;
+
+  private static final AttributePlainEnumValue ENUM_VALUE_A =
+      AttributePlainEnumValueBuilder.of().key("a").label("label_a").build();
+  private static final AttributePlainEnumValue ENUM_VALUE_B =
+      AttributePlainEnumValueBuilder.of().key("b").label("label_b").build();
+
+  private static final AttributeLocalizedEnumValue LOCALIZED_ENUM_VALUE_A =
+      AttributeLocalizedEnumValueBuilder.of().key("a").label(ofEnglish("label_a")).build();
+  private static final AttributeLocalizedEnumValue LOCALIZED_ENUM_VALUE_B =
+      AttributeLocalizedEnumValueBuilder.of().key("b").label(ofEnglish("label_b")).build();
+
+  /** Initialises test data. */
+  @BeforeAll
+  static void setup() {
+    old =
+        AttributeDefinitionBuilder.of()
+            .name("attributeName1")
+            .label(ofEnglish("label1"))
+            .type(AttributeTypeBuilder::textBuilder)
+            .isRequired(false)
+            .attributeConstraint(AttributeConstraintEnum.SAME_FOR_ALL)
+            .inputTip(ofEnglish("inputTip1"))
+            .inputHint(TextInputHint.TextInputHintEnum.SINGLE_LINE)
```

Comment:
This doesn't exist??
```suggestion
            .inputHint(TextInputHint.SINGLE_LINE)
```

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/989/commits/75b1adf883bf06918a01b0fda85d961c94d8e5da